src/atoms/cc-select.js

```diff
@@ -0,0 +1,217 @@
+import { css, html, LitElement } from 'lit-element';
+import { classMap } from 'lit-html/directives/class-map.js';
+
+import { dispatchCustomEvent } from '../lib/events.js';
+import { i18n } from '../lib/i18n.js';
+
+/**
+ * @typedef {import('./types.js').Option} Option
+ */
+
+/**
+ * A component wrapping native select element and its label.
+ *
+ * @cssdisplay inline-block
+ *
+ * @event {CustomEvent<string>} cc-select:input - Fires the `value` whenever the `value` changes.
+ *
+ * @slot error - The error message to be displayed right below the select element (low margin + red text)
+ */
+export class CcSelect extends LitElement {
+  static get properties () {
+    return {
+      disabled: { type: Boolean, reflect: true },
+      /** @required */
+      label: { type: String },
+      name: { type: String, reflect: true },
+      options: { type: Array },
+      placeholder: { type: String },
+      required: { type: Boolean },
+      value: { type: String },
+      _uniqueName: { type: String, attribute: false },
+      _uniqueId: { type: String, attribute: false },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {boolean} Sets `disabled` attribute on inner native `<select>` element. */
+    this.disabled = false;
+
+    /** @type {string|null} Sets `name` attribute on inner native `<select>` element. */
+    this.name = null;
+
+    /** @type {string|null} Sets label for the input. Mandatory but can be hidden if necessary. */
+    this.label = null;
+
+    /** @type {Option[]|[]} Sets the list of options displayed inside the select element. */
+    this.options = [];
+
+    /** @type {string|null} Sets a default option with empty value. */
+    this.placeholder = null;
+
+    /** @type {boolean} Sets the "required" text inside the label */
+    this.required = false;
+
+    /** @type {string|null} Sets the selected value of the element. */
+    this.value = null;
+
+    // use this unique id for isolation (Safari seems to have a bug)
+    /** @type {string} */
+    this._uniqueId = Math.random().toString(36).slice(2);
+
+    // use this unique name for isolation (Safari seems to have a bug)
+    /** @type {string} */
+    this._uniqueName = Math.random().toString(36).slice(2);
+  }
+
+  /**
+   * Triggers focus on the inner `<select>` element.
+   */
+  focus () {
+    this.shadowRoot.querySelector('select').focus();
+  }
+
+  _onSelectInput (e) {
+    this.value = e.target.value;
+    dispatchCustomEvent(this, 'input', this.value);
+  }
+
+  render () {
+    return html`
+      <label for=${this._uniqueName}>
+        ${this.label}
+        ${this.required ? html`
+          <span class="required">${i18n('cc-select.required')}</span>
+        ` : ''}
+      </label>
+      <div class="selectWrapper ${classMap({ disabled: this.disabled })}">
+        <select
+          id=${this._uniqueName}
+          ?disabled=${this.disabled}
+          aria-describedby=${this._uniqueErrorId}
```

Comment:
This was kept as `_uniqueErrorId` which would fail in the current state of the code. If we rename the other parts as described, this can be kept as is.